### DISCUSSION TITLE
Dot syntax for placeholder/read-only values

### DISCUSSION THREAD

**Discussion by kjlloydie** [filamentphp/filament#2135] [Ideas]
The dot syntax for accessing fields on a relation works for table fields, like this:
```php
use Filament\Tables\Columns\TextColumn;
  
TextColumn::make('author.name')
```
but it doesn't work if we wanted to display the data on a form in a (read-only) TextInput:
```php
use Filament\Forms\Components\TextInput

TextInput::make('author.name') // will be blank
  ->disabled(true)
```
Therefore, to show read-only data from a relation on a form, we have to use a Placeholder field:
```php
Placeholder::make("Author Name")
  ->content(function (?Model $record) {
      return ($record->author) ? $record->author->name : '';
 })
```
It would be great if there was a simpler/more consistent way of displaying this data, using dot syntax, either in a Placeholder or a read-only TextInput.

**Comment by ryangjchandler** [comment]
Forms (in the admin) are designed to operate on a single model / resource right now.

**Comment by danharrin** [comment]
To make dot-syntax work for readonly form fields, you just need to eager-load the relationship on `$this->record`, in the `beforeFill()` hook of the resource page.

**Reply by kokoshneta** [reply]
Is it still the case that this should work? Is eager-loading not just a case of using `load()` in the hook? I can’t seem to get it to work, at any rate.

For example, I have a `BookResource` page where I want to show the book’s DOI as a read-only property (i.e., it should not be editable, and updating the book resource should never insert a new row in the DOI table, even if there is no row for the book record being edited). If I do this:

```php
// EditBook.php
protected function beforeFill() {
  $this->record->load('doi');
}

// BookResource.php
public static function form(Form $form) : Form {
  return $form->schema([
    TextInput::make('doi.doi')
      ->disabled()
      ->label('DOI')
  ]);
}
```

– then the text input field is empty.

Modifying the resource’s base query like this also has no effect – the disabled text input field remains empty:

```php
// BookResource.php
public static function getEloquentQuery(): Builder {
  return parent::getEloquentQuery()
    ->with('doi')
  ;
}
```

If I add an empty `Grid` component somewhere in the `form()` method and specify the relationship on that, the value is fetched and displayed correctly. But then updating the book resource will also attempt to insert a row in the DOI table if there isn’t one already.

Am I doing something wrong?

**Reply by danharrin** [reply]
You shouldn't need to eager load anything, try this:


```php
Group::make([
    TextInput::make('doi')
      ->disabled()
      ->label('DOI')
])->relationship('doi')
```

**Reply by kokoshneta** [reply]
Oops, commented too fast. Accidentally tested it on a record that _has_ an existing DOI relationship, where it works; but testing on a record that doesn’t, it doesn’t work.

It works the same as if I add a separate `Grid` or `Group` component somewhere else and specify the relationship there (which I’d already tried):

- if the book resource **has** existing relationship data (i.e., there is a corresponding row in the DOI table), it loads and displays the relationship data correctly, but it _also_ updates the relationship when updating the base model
- if the book resource **doesn’t have** any existing relationship data (no corresponding row in the DOI table), it correctly leaves the form field empty, but when saving the book resource, it ‘updates’ the relationship by attempting to insert a new row into the DOI table

The purpose here is to **only** display the relationship data – never attempt to update it or add new rows. That’s why I wanted to try to eager-load the relationship: I was assuming that would make it available for dot syntax without registering a relationship that would be updated when saving.

**Reply by danharrin** [reply]
Use the group and then pass `->saveRelationshipsUsing(null)`

**Reply by kokoshneta** [reply]
Excellent – that worked! Thank you!

**Reply by kokoshneta** [reply]
Not sure how or why that worked initially, but it stopped working fairly quickly. I [finally just discovered the fix](https://github.com/filamentphp/filament/discussions/6378), which is to include a false boolean as the second parameter (`$saveCondition`) to `->relationship('doi', FALSE)`.
